data-prepper-api/src/main/java/org/opensearch/dataprepper/typeconverter/StringConverter.java

```diff
@@ -25,6 +34,11 @@ public String convert(Object source) throws IllegalArgumentException {
         if (source instanceof String) {
             return (String)source;
         }
-        throw new IllegalArgumentException("Unsupported type conversion");
+        LOG.error(EVENT, "Unable to convert {} to String", source);
+        if (Objects.nonNull(source)) {
+            return source.toString();
```

Comment:
I'm not sure this is really a good behavior. It could result in strange output. Say a list is provided. The user will get a string that looks like `ListString@89879` or something like that.

I think this should continue to throw the exception.

We can handle the exception in the `ConvertEntryTypeProcessor` class to make it consistent across all types. 

Perhaps by dropping the data?